sim/aviron/src/lib/Cpu.zig

```diff
@@ -109,20 +101,82 @@ pub fn dump_system_state(cpu: *Cpu) void {
     std.debug.print("Y (r29:r28): 0x{X:0>4}\n", .{y_reg});
     std.debug.print("Z (r31:r30): 0x{X:0>4}\n", .{z_reg});
 
-    // Dump SRAM
-    std.debug.print("\nSRAM DUMP (0x0100-0x08FF, {} bytes):\n", .{cpu.sram.size});
-    for (0..cpu.sram.size) |i| {
-        if (i % 16 == 0) {
-            std.debug.print("0x{X:0>4}: ", .{0x0100 + i});
+    // Dump all registers
+    std.debug.print("\nREGISTERS:\n", .{});
+    for (cpu.regs, 0..) |reg, i| {
+        if (i % 8 == 0) std.debug.print("r{d:0>2}-r{d:0>2}: ", .{ i, @min(i + 7, 31) });
+        std.debug.print("{X:0>2} ", .{reg});
+        if ((i + 1) % 8 == 0) std.debug.print("\n", .{});
+    }
+
+    // Dump SRAM using absolute addresses based on configured base
+    const sram_base: u24 = cpu.sram.get_base();
```

Comment:
This is the only use of `get_base`, which is accessed through the interface (a `RAM` object). We need the field in the `Static` struct (to know how to translate an address into an array offset), and so I needed to add this to the vtable to make it available here. I *could* also copy the base into the base struct when `memory()` is called, which would clean up the interface, but I am not sure what you think is nicer, so please weigh in.